src/components/tools/base.ts

```diff
@@ -226,10 +252,31 @@ export default abstract class BaseTool<Type extends Tool> {
     return this.constructable[CommonInternalSettings.SanitizeConfig];
   }
 
+  /**
+   * Returns true if Tools is inline
+   */
+  public isInline(): this is InlineTool {
+    return this.type === ToolType.Inline;
+  }
+
+  /**
+   * Returns true if Tools is block
+   */
+  public isBlock(): this is BlockTool {
```

Comment:
maybe it is better to rename this method to `isBlockTool`? because Block is a product of a tool

Reply:
Usage would be `tool.isBlock()`, so I think it's ok